android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIActivityIndicator.java

```diff
@@ -149,35 +128,47 @@ public void hide()
 		visible = false;
 	}
 
-	protected int getStyle()
-	{
-		if (proxy.hasProperty(TiC.PROPERTY_STYLE)) {
-			int style = TiConvert.toInt(proxy.getProperty(TiC.PROPERTY_STYLE));
-			if (style != PLAIN && style != BIG && style != DARK && style != BIG_DARK) {
-				Log.w(TAG, "Invalid value \"" + style + "\" for style.");
-				return PLAIN;
-			}
-			return style;
-		}
-		return PLAIN;
-	}
-
-	protected void setStyle(int style)
+	private void updateIndicator()
 	{
-		if (style == currentStyle) {
+		// Do not continue if proxy has been released.
+		if (this.proxy == null) {
 			return;
 		}
-		if (style != PLAIN && style != BIG && style != DARK && style != BIG_DARK) {
-			Log.w(TAG, "Invalid value \"" + style + "\" for style.");
-			return;
+
+		// Fetch assigned style ID.
+		int styleId = TiConvert.toInt(this.proxy.getProperty(TiC.PROPERTY_STYLE), PLAIN);
+		if ((styleId != PLAIN) && (styleId != BIG) && (styleId != DARK) && (styleId != BIG_DARK)) {
+			Log.w(TAG, "Invalid value \"" + styleId + "\" for style.");
+			styleId = PLAIN;
 		}
-		LinearLayout view = (LinearLayout) getNativeView();
 
-		view.removeAllViews();
-		progress = new ProgressBar(TiApplication.getAppCurrentActivity(), null, style);
-		currentStyle = style;
-		view.addView(progress);
-		view.addView(label);
-		view.requestLayout();
+		// Update indicator to use a big or small style.
+		int[] idArray = new int[] {
+			R.attr.trackThickness,
+			R.attr.indicatorSize,
+			R.attr.indicatorInset
+		};
+		int themeId = R.style.Widget_MaterialComponents_CircularProgressIndicator_ExtraSmall;
+		if ((styleId == BIG) || (styleId == BIG_DARK)) {
+			themeId = R.style.Widget_MaterialComponents_CircularProgressIndicator_Medium;
+		}
+		ContextThemeWrapper context = new ContextThemeWrapper(this.progress.getContext(), themeId);
+		TypedArray typedArray = context.obtainStyledAttributes(null, idArray, 0, 0);
+		int value = typedArray.getDimensionPixelSize(0, this.progress.getTrackThickness());
+		this.progress.setTrackThickness(typedArray.getDimensionPixelSize(0, this.progress.getTrackThickness()));
+		this.progress.setIndicatorSize(typedArray.getDimensionPixelSize(1, this.progress.getIndicatorSize()));
+		this.progress.setIndicatorInset(typedArray.getDimensionPixelSize(2, this.progress.getIndicatorInset()));
+		typedArray.recycle();
+
+		// Update indicator's color.
+		if (this.proxy.hasPropertyAndNotNull(TiC.PROPERTY_INDICATOR_COLOR)) {
+			int color = TiConvert.toColor(TiConvert.toString(this.proxy.getProperty(TiC.PROPERTY_INDICATOR_COLOR)));
+			this.progress.getIndeterminateDrawable().setColorFilter(color, android.graphics.PorterDuff.Mode.SRC_IN);
```

Comment:
Looks like this usage of `setColorFilter()` was deprecated in API 29, maybe we should have a clause here to handle >29?

Reply:
Good catch. There is another `setColorFilter(ColorFilter)` method that I can use and it's been available since API Level 1. I'll switch it to that. Thanks.